src/tools/run-make-support/src/rustc.rs

```diff
@@ -4,14 +4,17 @@ use std::io::Write;
 use std::path::Path;
 use std::process::{Command, Output, Stdio};
 
+use crate::drop_bomb::DropBomb;
 use crate::{handle_failed_output, set_host_rpath, tmp_dir};
 
 /// Construct a new `rustc` invocation.
+#[must_use]
 pub fn rustc() -> Rustc {
```

Comment:
We could just mark `Rustc` with `#[must_use]`, to avoid forgetting to put these annotations on top of functions that create `Rustc`.

Reply:
Good point, I moved `#[must_use]` annotations to the command wrapper structs themselves.